### PR TITLE
Make sure welcome screen is closed

### PR DESCRIPTION
**Pull request by ggardet**
- Related ticket: https://progress.opensuse.org/issues/56024
- Verification run: https://openqa.opensuse.org/tests/1020352

### CODE REVIEW

**Comment by Vogtinator** [comment]
Is it enough to just assert_screen an unticked checkbox after the first click?

IMO it should softfail if one click is not enough.

**Comment by ggardet** [comment]
> Is it enough to just assert_screen an unticked checkbox after the first click?
> 
> IMO it should softfail if one click is not enough.

I think the real problem is the system is unresponsive for a while, so the clicks are just missed. So, IMO, no need to soft fail as it is not really a failure, just slowness.

**Comment by mloviska** [comment]
The notification icon can be activated by mouse hover? It's visible only from the video

**Comment by ggardet** [comment]
> The notification icon can be activated by mouse hover? It's visible only from the video

Maybe. Is it a problem?

File: tests/installation/opensuse_welcome.pm [hunk 1]
```diff
@@ -20,7 +20,32 @@ use utils;
...
     assert_screen("opensuse-welcome");
     assert_and_click("opensuse-welcome-show-on-boot");
+    # Make sure tick box is unchecked - poo#56024
+    for my $retry (1 .. 5) {
+        if (check_screen("opensuse-welcome-show-on-boot")) {
+            assert_and_click("opensuse-welcome-show-on-boot");
+        } else {
+            last;
+        }
+        if ($retry == 5) {
```
Comment (StefanBruens) [inline tests/installation/opensuse_welcome.pm]: If the click succeeds on the 5th try, the test still dies.

Better:
```
assert_screen("opensuse-welcome")
for my $retry (1..5) {
  assert_and_click("opensuse-welcome-show-on-boot");
  last unless check_screen("opensuse-welcome-show-on-boot");
  die "Unable to untick 'Show on next startup'" if $retry == 5;
}
```

**Comment by mloviska** [comment]
> > The notification icon can be activated by mouse hover? It's visible only from the video
> 
> Maybe. Is it a problem?

If it is not triggered by user interaction, it might be interesting to dive in deeper. It is always weird when a freshly installed system is laggy/glitchy/not responsive.

**Comment by ggardet** [comment]
> > > The notification icon can be activated by mouse hover? It's visible only from the video
> > 
> > 
> > Maybe. Is it a problem?
> 
> If it is not triggered by user interaction, it might be interesting to dive in deeper. It is always weird when a freshly installed system is laggy/glitchy/not responsive.

I think the lack of responsiveness is due to a load on host (probably due to the snapshot save done in previous step).

File: tests/installation/opensuse_welcome.pm (hunk 1 above)
Comment (ggardet) [reply tests/installation/opensuse_welcome.pm]: It should be ok now.

**Comment by okurz** [comment]
looks safe enough, let's try

**Comment by Vogtinator** [comment]
I have a suspicion what's going on: The mouse must not have moved a single pixel before the checkbox actually registers the mouse button up event. So maybe it's enough to just wait a bit before moving the cursor away again. I'll give it a try.

**Comment by okurz** [comment]
that sounds great. I would appreciate if you give it a try because it sounds like this could help us also in other cases to instrument properly

**Comment by Vogtinator** [comment]
> that sounds great. I would appreciate if you give it a try because it sounds like this could help us also in other cases to instrument properly

Tried, but unfortunately didn't work. What did work though was before the opensuse-welcome module ran, logging into VNC and restarting the opensuse-welcome application.

**Pull request merged**